src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -128,11 +128,22 @@ public void UpdateVariable(string name, TimeSpan value)
         }
 
         /// <summary>
-        /// Create or update a named variable to a value. 
+        /// Create or update a named variable to a value, with custom CanSet/Mutate attributes. 
         /// </summary>
         /// <param name="name">variable name. This can be used in other formulas.</param>
         /// <param name="value">constant value.</param>
         public void UpdateVariable(string name, FormulaValue value)
+        {
+            UpdateVariable(name, value, new SymbolProperties { CanMutate = true, CanSet = true });
+        }
+
+        /// <summary>
+        /// Create or update a named variable to a value, with custom CanSet/Mutate attributes. 
+        /// </summary>
+        /// <param name="name">variable name. This can be used in other formulas.</param>
+        /// <param name="value">constant value.</param>
+        /// <param name="newVarProps">symbol properties.</param>
+        public void UpdateVariable(string name, FormulaValue value, SymbolProperties newVarProps)

```

Comment:
>public void UpdateVariable(string name, FormulaValue value, SymbolProperties newVarProps) [](http://example.com/codeflow?start=8&length=89)

I see the only place to leverage CanSetMutate prop is here,
How would I leverage it at compile time? If I was just doing engine.check() 
(At that time formula value doesn't exist)

Should we also expose this flag at symbolTable.AddVariable() ?

Reply:
AddVariable already has an overload that takes SymbolProperties as an argument in SymbolTable.cs line 119-ish:
```
public ISymbolSlot AddVariable(string name, FormulaType type, SymbolProperties props, string displayName = null)
```